linux/keyman-config/keyman_config/uninstall_kmp.py

```diff
@@ -36,15 +36,26 @@ def uninstall_kmp_shared(packageID):
         packageID (str): Keyboard package ID
     """
     kbdir = get_keyboard_dir(InstallLocation.Shared, packageID)
+    kbdocdir = get_keyman_doc_dir(InstallLocation.Shared, packageID)
+    kbfontdir = get_keyman_font_dir(InstallLocation.Shared, packageID)
+
+    logging.info("Uninstalling shared keyboard: %s", packageID)
+
+    # need to uninstall from ibus for all lang and all kmx in kmp
+    info, system, options, keyboards, files = get_metadata(kbdir)
+    if keyboards:
+        if is_gnome_shell():
+            uninstall_keyboards_from_gnome(keyboards, kbdir)
+        else:
+            uninstall_keyboards_from_ibus(keyboards, kbdir)
+    else:
+        logging.warning("could not uninstall keyboards")
+
     if not os.path.isdir(kbdir):
         msg = _("Keyboard directory for %s does not exist." % packageID)
         logging.error(msg)
         return msg
```

Comment:
I see even if `kbdir` doesn't exist, the keyboards get uninstalled from `uninstall_keyboard_from_gnome` or `uninstall_keyboards_from_ibus`.

But if kbdir doesn't exist, won't l 57. exit? Then, how does the kbddocdir at l.68 or kbfontdir at l. 72 get removed?

Reply:
Good point. Did some more refactoring how the uninstallation is done.